src/components/externalLinks/ExternalLinks.component.tsx

```diff
@@ -0,0 +1,1030 @@
+                <h2 className="fs-16 fw-4 c-9">No matching results</h2>
+            </EmptyState.Title>
+            <EmptyState.Subtitle>We couldn't find any matching external link configuration</EmptyState.Subtitle>
+        </EmptyState>
+    )
+}
+
+const formatOptionLabelClusters = (option: OptionType): JSX.Element => {
+    return (
+        <div className="flex left column">
+            <span>{option.label}</span>
+            {option.value === '*' && (
+                <>
+                    <small className="cn-6">All existing and future clusters</small>
+                    <div className="modal__dropdown-divider" />
+                </>
+            )}
+        </div>
+    )
+}
+
+const getErrorLabel = (field: string): JSX.Element => {
+    const errorLabel = (label: string): JSX.Element => {
+        return (
+            <div className="error-label flex left align-start fs-11 mt-4">
+                <Error className="icon-dim-20" />
+                <div className="ml-4 cr-5">{label}</div>
+            </div>
+        )
+    }
+    switch (field) {
+        case 'tool':
+            return errorLabel('Please select monitoring tool.')
+        case 'name':
+            return errorLabel('Please provide name for the tool you want to link.')
+        case 'cluster':
+            return errorLabel('Please select one or more clusters.')
+        case 'url':
+            return errorLabel('Please enter URL template.')
+        default:
+            return <></>
+    }
+}
+
+const ConfigureLinkAction = ({
+    index,
+    link,
+    clusters,
+    selectedClusters,
+    monitoringTools,
+    showDelete,
+    onMonitoringToolSelection,
+    onClusterSelection,
+    onNameChange,
+    onUrlTemplateChange,
+    deleteLinkData,
+}: ConfigureLinkActionType): JSX.Element => {
+    return (
+        <div id={`link-action-${index}`} className="configure-link-action-wrapper">
+            <div className="link-monitoring-tool mb-8">
+                <div className="monitoring-tool mr-8">
+                    <span>Monitoring Tool*</span>
+                    <Select
+                        key={`monitoring-tool-${index}`}
+                        placeholder="Select tool"
+                        name={`monitoring-tool-${index}`}
+                        value={link.tool}
+                        options={monitoringTools}
+                        isMulti={false}
+                        hideSelectedOptions={false}
+                        onChange={(selected) => onMonitoringToolSelection(index, selected)}
+                        components={{
+                            IndicatorSeparator: null,
+                            ClearIndicator: null,
+                            Option: OptionWithIcon,
+                            ValueContainer: ValueContainerWithIcon,
+                        }}
+                        styles={{
+                            ...multiSelectStyles,
+                            container: (base, state) => ({
```

Comment:
can we pull out common properties so that we can reuse it.

Reply:
Done. Moved common styles to utils.